### PR TITLE
#4507 add select network screen after you create import a new wallet

### LINKED ISSUE #4507: Add "Select Network" screen after you create/import a new wallet
**Problem:**
During user tests in Amsterdam, I found that users don't really know how to enable their favorite networks. They don't even understand that we support multiple networks.

**Solution:**
Add one more screen where they can select multiple networks from the existing base (we make a long list of all 32 supported networks). If they would need a custom one, then they could do this later. 

**Why?**
- first, we let users know we support multiple chains
- the wallet is already customized, so they don't have to go to the Settings tab

Invision Prototype:
https://invis.io/RA12NU40P8SW

select-network-flow

### PR DESCRIPTION
**Pull request by eviltofu**
Closes #4507.

I will squish and merge once approved.

##### Create Wallet Test:
1. Delete the app from your device.
2. Run.
3. Select "Create wallet" (screen 7).
4. Expect InitialNetworkSelectionViewController (screen 1) to show up.
5. Select one or more networks in main net.
6. Press continue button.
7. Expect TokensViewController (screen 6) to show up.

##### Import Wallet Test:
1. Delete the app from your device.
2. Run.
3. Select "I already have a wallet" (screen 7).
4. Select "Import existing wallet" (screen 8).
5. Enter wallet data.
6. Expect InitialNetworkSelectionViewController (screen 1) to show up.
7. Select one or more networks in mainnets.
8. Press continue button.
9. Expect TokensViewController (screen 6) to show up. 

#### Watch Wallet Test:
1. Delete the app from your device.
2. Run.
3. Select "I already have a wallet" (screen 7).
4. Select "Track wallet address" (screen 8).
5. Enter wallet address.
6. Expect TokensViewController (screen 6) to show up. 

#### Testnet Test:
1. Delete the app from your device.
2. Run.
3. Select "Create wallet" (screen 7).
4. Expect to see InitialNetworkSelectionViewController (screen 1).
5. Click on the Mainnet toggle switch to turn off mainnets.
6. Expect Overlay to appear (screen 3).
7. Press "Got it, enable testnets" button.
8. Expect Testnets screen (screen 4).
9. Click on the Testnet toggle switch to turn off testnets.
10. Expect to see InitialNetworkSelectionViewController (screen 1).

#### Filter Test:
1. Delete the app from your device.
2. Run.
3. Select "Create wallet" (screen 7).
4. Expect to see InitialNetworkSelectionViewController (screen 1).
5. Select the searchBar and type some letters.
6. Expect to see searchBar filtering (screen 2) or empty search results (screen 9).

#### Continue button Test:
1. Delete the app from your device.
2. Run.
3. Select "Create wallet" (screen 7).
4. Expect to see InitialNetworkSelectionViewController (screen 1).
5. Deselect all networks.
6. Expect to see Continue button disabled (screen 5)

| Screen # | 
| ----- |
| Screen 1<BR>Screen 1<BR>InitialNetworkSelectionViewController |
| <BR>Screen 2<BR>SearchBar filtering |
| <BR>Screen 3<BR>Testnets Overlay |
| <BR>Screen 4<BR>Testnets |
| <BR>Screen 5<BR>Continue button disabled when no networks are selected |
| <BR>Screen 6<BR>TokensViewController |
| <BR>Screen 7<BR>Initial Wallet Creation |
| <BR>Screen 8<BR>Import/Track Wallet Options |
| <BR>Screen 9<BR>Empty search results |

### CODE REVIEW

**Comment by eviltofu** [comment]
Ack! I need to implement the empty screen when the search filter turns up nothing.

**Comment by hboon** [comment]
Can just skip it and do it in another PR. If reasonable, always merge a smaller PR.

**Review by hboon — CHANGES_REQUESTED** [review]
- [x] Switching to testnets should have some defaults like in Settings > Select Active Networks.

- [x] Switching to testnets and selecting some still only enables the default mainnets:

    1. Delete app
    2. Build and install
    3. Create wallet
    4. Switching to testnets
    5. Tap to enable Ropsten
    6. Continue

    Expected:

    Wallet tab shows Ropsten

    Observed:

    Wallet tab shows Ethereum, Gnosis and Polygon enabled (instead of Ropsten testnet)

    Settings > Select Active Networks shows these mainnets too.

File: AlphaWallet/Wallet/ViewModels/InitialNetworkSelectionViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        Array(model.selected)
+    }
+
+    init(model: InitialNetworkSelectionCollectionModel) {
+        self.model = model
+        super.init()
+        configureViewModel()
+    }
+
+    private func configureViewModel() {
```
Comment (hboon) [inline AlphaWallet/Wallet/ViewModels/InitialNetworkSelectionViewModel.swift:32]: Can just call it `configure()`

**Review by hboon — CHANGES_REQUESTED** [review]

**Comment by eviltofu** [comment]
The current master f8a81891fe88186401f131890fefc560a455b597 has the following behaviour:

If I make changes in the main nets, then return to the main screen, I get the main screen. I cannot select no active networks. Selecting no active networks and returning to the main screen does not give an error message; the app simply does nothing. The app should inform the user.

Simulator Screen Recording - iPhone 11 - 2022-05-17 at 10 09 09

@hboon

**Comment by eviltofu** [comment]
I'm going to add the following behaviour:

1) If main nets or tests nets are selected and there are no active selected networks, a default set of selected networks will be presented.
2) When no networks are selected in the current screen (main net or test net), the user cannot hit the continue button.
3) The current selected screen (main net or test net) will have their selected networks loaded. The selected networks in the other screen (test net or main net) will be ignored.

**Comment by hboon** [comment]
> If I make changes in the main nets, then return to the main screen, I get the main screen. Selecting no active networks and returning to the main screen does not give an error message; the app simply does nothing. The app should inform the user.

I'm not sure if you decided on something different in the earlier comment, if so ignore this:

I think it's good to just go back and do nothing if the user de-selects everything. It's a less hassle than the user trying to remember what they want to select, and less disruptive than a popup, but not too important.

**Comment by hboon** [comment]
> 1. If main nets or tests nets are selected and there are no active selected networks, a default set of selected networks will be presented.

This is right after creating a wallet right? Sounds OK.

> 2. When no networks are selected in the current screen (main net or test net), the user cannot hit the continue button.

OK, but this contradicts (1) above

> 3. The current selected screen (main net or test net) will have their selected networks loaded. The selected networks in the other screen (test net or main net) will be ignored when the continue button is pressed.

Not sure what this means.

> Should the user not be allowed to deselect the last selected network?

They should be allowed to with the expectation that they go and select another network. Otherwise it'd be annoying and confusing.

**Review by hboon — APPROVED** [review]
Looks good.

**Comment by eviltofu** [comment]
> > 1. If main nets or tests nets are selected and there are no active selected networks, a default set of selected networks will be presented.
> 
> This is right after creating a wallet right? Sounds OK.
> 
> > 2. When no networks are selected in the current screen (main net or test net), the user cannot hit the continue button.
> 
> OK, but this contradicts (1) above

The user can deselect networks.
> 
> > 3. The current selected screen (main net or test net) will have their selected networks loaded. The selected networks in the other screen (test net or main net) will be ignored when the continue button is pressed.
> 
> Not sure what this means.

If you are in test net, only test nets will be added. If you are in main nets, only main nets will be added. 

> 
> > Should the user not be allowed to deselect the last selected network?
> 
> They should be allowed to with the expectation that they go and select another network. Otherwise it'd be annoying and confusing.

**Comment by hboon** [comment]
> I will squish and merge once approved.

Eek. That didn't happen.

**Pull request merged**